crates/ol/rpc/api/src/types.rs

```diff
@@ -0,0 +1,260 @@
+
+impl From<AccountId> for RpcAccountId {
+    fn from(id: AccountId) -> Self {
+        RpcAccountId(*id.inner())
+    }
+}
+
+impl From<RpcAccountId> for AccountId {
+    fn from(rpc: RpcAccountId) -> Self {
+        AccountId::new(rpc.0)
+    }
+}
+
+/// OL chain status with latest, confirmed, and finalized blocks.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct RpcOlChainStatus {
+    pub latest: OLBlockCommitment,
+    pub confirmed: OLBlockCommitment,
+    pub finalized: OLBlockCommitment,
+}
+
+/// Message payload with bitcoin value and data.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct RpcMsgPayload {
+    /// Value in satoshis.
+    pub value: u64,
+    /// Hex-encoded data.
+    #[serde(with = "hex::serde")]
+    pub data: Vec<u8>,
+}
+
+impl From<MsgPayload> for RpcMsgPayload {
+    fn from(payload: MsgPayload) -> Self {
+        Self {
+            value: payload.value().to_sat(),
+            data: payload.data().to_vec(),
+        }
+    }
+}
+
+impl From<RpcMsgPayload> for MsgPayload {
+    fn from(rpc: RpcMsgPayload) -> Self {
+        MsgPayload::new(BitcoinAmount::from_sat(rpc.value), rpc.data)
+    }
+}
+
+impl From<&MsgPayload> for RpcMsgPayload {
+    fn from(payload: &MsgPayload) -> Self {
+        Self {
+            value: payload.value().to_sat(),
+            data: payload.data().to_vec(),
+        }
+    }
+}
+
+/// Message entry with source account, epoch, and payload.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct RpcMessageEntry {
+    #[serde(with = "hex::serde")]
+    pub source: [u8; 32],
+    pub incl_epoch: u32,
+    pub payload: RpcMsgPayload,
+}
+
+/// Proof state: inner state commitment and next message index.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct RpcProofState {
+    #[serde(with = "hex::serde")]
+    pub inner_state: [u8; 32],
+    pub next_inbox_msg_idx: u64,
+}
+
+/// Update state data with proof state and extra data.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct RpcUpdateStateData {
+    pub proof_state: RpcProofState,
+    #[serde(with = "hex::serde")]
+    pub extra_data: Vec<u8>,
+}
+
+/// Update input data for state transitions.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct RpcUpdateInputData {
+    pub seq_no: u64,
+    pub messages: Vec<RpcMessageEntry>,
+    pub update_state: RpcUpdateStateData,
+}
+
+/// Transaction payload: generic message or snark account update.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(tag = "type", rename_all = "snake_case")]
+pub enum RpcTransactionPayload {
+    GenericAccountMessage {
+        #[serde(with = "hex::serde")]
+        target: [u8; 32],
+        #[serde(with = "hex::serde")]
+        payload: Vec<u8>,
+    },
+    SnarkAccountUpdate {
+        #[serde(with = "hex::serde")]
+        target: [u8; 32],
+        update: RpcUpdateInputData,
+        #[serde(with = "hex::serde")]
+        update_proof: Vec<u8>,
+    },
```

Comment:
Struct-like enum variants are annoying to use.  Split these out as their own structs.